src/plugins/console/public/application/models/legacy_core_editor/mode/output_highlight_rules.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import ace from 'brace';
+import 'brace/mode/json';
+import { addXJsonToRules } from '@kbn/ace';
+
+const JsonHighlightRules = ace.acequire('ace/mode/json_highlight_rules').JsonHighlightRules;
+
+export class OutputJsonHighlightRules extends JsonHighlightRules {
+  constructor() {
+    super();
+    this.$rules = {};
+    addXJsonToRules(this, 'start');
+    this.$rules.start.unshift(
+      {
+        token: 'warning',
+        regex: '#!.*$',
+      },
+      {
+        token: 'comment',
+        regex: /#(.*?)(?=\d+\s(?:[\sA-Za-z]+)|$)/,
+      },
+      {
+        token(value: string) {
```

Comment:
Can we change this line to `token: mapStatusCodeToBadge,` and extract this function onto line 15:

```ts
export const mapStatusCodeToBadge = (value: string) => {
  const regExpMatchArray = value.match(/\d+/);
  if (regExpMatchArray) {
    const status = parseInt(regExpMatchArray[0], 10);
    if (status <= 199) {
      return 'badge.badge--default';
    }
    if (status <= 299) {
      return 'badge.badge--success';
    }
    if (status <= 399) {
      return 'badge.badge--primary';
    }
    if (status <= 499) {
      return 'badge.badge--warning';
    }
    return 'badge.badge--danger';
  }
};
```

Then could you create a sibling file `output_highlight_rules.test.ts` and add these tests? This way if someone makes changes to this logic the tests will catch it. Thanks!

```ts
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { mapStatusCodeToBadge } from './output_highlight_rules';

describe('mapStatusCodeToBadge', () => {
  const testCases = [{
    description: 'treats 100 as as default',
    value: '# PUT test-index 100 Continue',
    badge: 'badge.badge--default',
  }, {
    description: 'treats 200 as success',
    value: '# PUT test-index 200 OK',
    badge: 'badge.badge--success',
  }, {
    description: 'treats 301 as primary',
    value: '# PUT test-index 301 Moved Permanently',
    badge: 'badge.badge--primary',
  }, {
    description: 'treats 400 as warning',
    value: '# PUT test-index 404 Not Found',
    badge: 'badge.badge--warning',
  }, {
    description: 'treats 502 as danger',
    value: '# PUT test-index 502 Bad Gateway',
    badge: 'badge.badge--danger',
  }, {
    description: 'treats unexpected numbers as danger',
    value: '# PUT test-index 666 Demonic Invasion',
    badge: 'badge.badge--danger',
  }, {
    description: 'treats no numbers as undefined',
    value: '# PUT test-index',
    badge: undefined,
  }];

  testCases.forEach(({ description, value, badge }) => {
    test(description, () => {
      expect(mapStatusCodeToBadge(value)).toBe(badge);
    });
  });
});
```

Reply:
Sure, thanks for the suggestion!